### PR TITLE
fix attention mask

### PR DESCRIPTION
**Pull request by nguyenhoangthuan99**
fix attention mask issue https://github.com/menloresearch/jan-models/issues/94#issuecomment-3148906493

### CODE REVIEW

**Review by gau-nernst — APPROVED** [review]
Since you are already adding this, should we
- Rename current `attention_mask` to `loss_mask`
- Rename the new `attention_mask_for_forward` to `attention_mask`

But maybe the rename will introduce some bugs in other parts of the codebase tho 🤣. safer to do like this....
